src/components/app/index.tsx

```diff
@@ -2,17 +2,58 @@ import { h, Component } from 'preact';
 import { bind, bitmapToImageData } from '../../lib/util';
 import * as style from './style.scss';
 import Output from '../output';
+import Options from '../options';
 
-import {MozJpegEncoder} from '../../lib/codec-wrappers/mozjpeg-enc';
+import { Encoder } from '../../lib/codec-wrappers/codec';
+import { MozJpegEncoder } from '../../lib/codec-wrappers/mozjpeg-enc';
+
+export type ImageType = 'original' | 'jpeg';
+
+export type CodecOptions = any;
+
+type Encoders = {
+  [type: string]: new () => Encoder
+};
+
+const EncoderNames = {
+  original: 'Original Image',
```

Comment:
Make this a function that just always returns a string to have a consistent type for this dictionary?

Reply:
I'm fine with that.